pkg/scheduler/objects/node_collection.go

```diff
@@ -211,13 +213,7 @@ func (nc *baseNodeCollection) NodeUpdated(node *Node) {
 	nc.Lock()
 	defer nc.Unlock()
 
-	nref := nc.nodes[node.NodeID]
-	if nref == nil {
-		return
-	}
-
-	updatedScore := nc.scoreNode(node)
-	if nref.nodeScore != updatedScore {
+	if nref, ok := nc.nodes[node.NodeID]; ok {
```

Comment:
I don't think this is correct. When NodeUpdated() is called, it might be due to usage values changing, and therefore we need to calculate a new score and update nref.nodeScore. If the score has changed, the node needs to be removed from the BTree (looked up using previous version of nref) and re-added using the updated nref.

Reply:
Actually, I may be mistaken. It looks like you're unconditionally removing and re-adding the node? If so, that also works.

Reply:
>  It looks like you're unconditionally removing and re-adding the node?

yep. That may be a small performance issue since it removes and re-add the node even if the resources are not changed. However, the code of checking the changes gets more complicated after the type is changed from `float64` to `[]float64`. I prefer to update node unconditionally to keep code simpler.

Reply:
Makes sense.